__tests__/main.test.ts

```diff
@@ -286,6 +287,15 @@ describe("VIB", () => {
     )
   })
 
+  it("Reads a pipeline and validates its functionality", async () => {
```

Comment:
You need to add another test that proves the opposite. Sending a bad pipeline and proving the action fails.